core/post.ts

```diff
@@ -94,6 +104,73 @@ export function getPostIntro(post: OriginPost): string {
   }
 }
 
+export async function calcPostReading(
+  post: OriginPost,
+  feed?: FeedValue,
+  filterId?: string
+): Promise<'fast' | 'slow'> {
+  let checker = prepareFilters(
+    await loadFilters({
+      feedId: feed?.id,
+      id: filterId
+    })
+  )
+  let action = checker(post)
+
+  if (action) {
+    return action === 'fast' ? 'fast' : 'slow'
+  }
+  if (feed) {
+    return feed?.reading ?? 'fast'
+  }
+  return 'fast'
+}
+
+export async function getPostsByFilter(filterId: string): Promise<PostValue[]> {
+  let filters = await loadFilters({ id: filterId })
+
+  if (filters.length === 0) return []
+
+  let checker = prepareFilters(filters)
+  let filterPosts = await loadPosts()
+  filterPosts = filterPosts.filter(post => checker(post))
+
+  return filterPosts
+}
+
+export async function changePostsByFeed(feedId: string): Promise<void> {
+  let posts = await loadPosts({ feedId })
+  let feed = await loadValue(await getFeed(feedId))
+  for (let post of posts) {
+    let reading = await calcPostReading(post, feed)
+    await changePost(post.id, { reading })
+  }
+}
+
+export async function changePostsByFilter(filterId: string): Promise<void> {
```

Comment:
Do we really need separated function for `changePostsByFeed` and `changePostsByFilter`?

In all cases we need to:
1. load all filters (because filters order is important)
2. load all posts
3. re-calc post type according to filters and default type

I suggest create functions like `recalcPostsReading(feedId)`.